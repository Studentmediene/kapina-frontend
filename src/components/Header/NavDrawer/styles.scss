@import 'variables.scss';

.navDrawer {
  position: fixed;
  top: 0;
  left: 0;
  height: 100vh;
  width: 290px;
  max-width: 0;
  z-index: 110;
  overflow: hidden;

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;


  cursor: initial;
  background-color: $color-radiorevolt-lightgrey;
  box-shadow: 2px 0 2px $color-radiorevolt-darkgrey;
  transition: max-width 0.3s;
}

.open {
  max-width: 100vw;;
}

.logoRow {
  display: flex;
  flex-direction: row;
  justify-content: center;
  
  width: 100%;
  padding: 10px;
}

.logo {
  height: 100px;
}

.navItem {
  width: 100%;
  padding: 20px 30px;
  font-size: 21px;
  border-top: 1px solid $color-radiorevolt-darkgrey;
  border-bottom: 1px solid $color-radiorevolt-darkgrey;

  &:hover {
    cursor: pointer;
    background-color: $color-radiorevolt-darkgrey;
  }
}