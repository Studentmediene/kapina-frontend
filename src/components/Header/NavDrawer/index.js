import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { withRouter } from 'next/router';

import styles from './styles.scss';

export class NavDrawer extends React.Component {
  static propTypes = {
    open: PropTypes.bool.isRequired,
    links: PropTypes.array.isRequired,
    onNavigation: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.navigate = this.navigate.bind(this);
    this.preventInteraction = this.preventInteraction.bind(this);
  }

  navigate(e, destination) {
    this.props.onNavigation(e);
    this.props.router.push(destination);
    // Stop browser from acting on the link click
    return false;
  }

  preventInteraction(e) {
    e.stopPropagation();
  }

  render() {
    const navbarComponents = links =>
      links.map(link => (
        <a
          href={link.path}
          key={link.path}
          className={styles.navItem}
          onClick={e => this.navigate(e, link.path)}
        >
          {link.title}
        </a>
      ));

    return (
      <div
        className={classNames({
          [styles.navDrawer]: true,
          [styles.open]: this.props.open,
        })}
        onClick={this.preventInteraction}
      >
        <a
          href="/"
          className={styles.logoRow}
          onClick={e => this.navigate(e, '/')}
        >
          <img src="/assets/RR_logo.png" className={styles.logo} />
        </a>
        {navbarComponents(this.props.links)}
      </div>
    );
  }
}

export default withRouter(NavDrawer);
