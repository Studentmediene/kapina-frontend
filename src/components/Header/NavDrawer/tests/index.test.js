import React from 'react';
import { shallow } from 'enzyme';

import NavDrawer from '../';

describe('<NavDrawer />', () => {
  it('renders correctly', () => {
    const props = {
      links: [
        {
          path: '/link1',
          title: 'Link1',
        },{
          path: '/link2',
          title: 'Link2',
        }
      ]
    }
    const tree = shallow(<NavDrawer {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
