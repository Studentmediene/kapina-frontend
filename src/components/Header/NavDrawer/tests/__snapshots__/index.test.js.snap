// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavDrawer /> renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;
